models/DataObject/ClassDefinition/Data.php

```diff
@@ -133,31 +133,34 @@
      * Returns the the data that should be stored in the resource
      *
      * @param mixed $data
+     * @param null|\Pimcore\Model\DataObject\AbstractObject $object
+     * @param mixed
      *
      * @return mixed
-     *
-     * abstract public function getDataForResource($data);
      */
+    abstract public function getDataForResource($data, $object = null, $params = []);
```

Comment:
I agree with you about that, but this shouldn't be in this PR since it doesn't belong to what to are trying to do